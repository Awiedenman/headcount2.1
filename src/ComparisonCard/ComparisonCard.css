.comparison-card{
  display: flex;
  justify-content: center;
  text-align: center;
}