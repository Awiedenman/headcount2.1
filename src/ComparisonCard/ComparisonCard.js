import React from 'react';
import './ComparisonCard.css';
import PropTypes from 'prop-types';

export const ComparisonCard = ({ 
  clickedCards, 
  compareDistrictAverages
}) => {
  
  if (clickedCards.length > 1) {
    const comparedDistrictsObj = 
    compareDistrictAverages(
      clickedCards[0].location,
      clickedCards[1].location
    );

    return (
      <div className='card clicked comparison-card'>
        <h2
          className="compare-location1">{
            clickedCards[0].location + ":" + 
            comparedDistrictsObj[clickedCards[0].location.toUpperCase()] }
        </h2>
        <h2>{`<------ ${comparedDistrictsObj.compared} ------>`}</h2>
        <h2>{ clickedCards[1].location + ":" + 
          comparedDistrictsObj[clickedCards[1].location.toUpperCase()] }
        </h2>
      </div>
    );   
  } else {
    return <div></div>;
  }
};

ComparisonCard.propTypes = {
  clickedCards: PropTypes.array,
  compareDistrictAverages: PropTypes.func,
  findAverage: PropTypes.func
};
