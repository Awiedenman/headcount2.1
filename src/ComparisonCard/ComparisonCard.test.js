import React from 'react';
import { shallow } from 'enzyme';
import { ComparisonCard } from './ComparisonCard';

describe('Comparison Card', () => {
  const mockRenderClickedCards = jest.fn();
  const mockCompareDistrictAverages = jest.fn();
  const mockFindAverage = jest.fn();
  it('should match snapshot upon render with two districts to compare', () => {
    const wrapper = shallow(<ComparisonCard
                            renderClickedCards={ mockRenderClickedCards }
                            compareDistrictAverages={ mockCompareDistrictAverages }
                            findAverage={ mockFindAverage }
                            />);
    expect(wrapper).toMatchSnapshot();
  })
})