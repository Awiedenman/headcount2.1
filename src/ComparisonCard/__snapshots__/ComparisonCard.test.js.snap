// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison Card should match snapshot upon render with two districts to compare 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    compareDistrictAverages={[Function]}
    renderClickedCards={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
